vowpalwabbit/accumulate.cc

```diff
@@ -69,38 +74,42 @@ float min_elem(float* arr, int length)
   return min;
 }
 
-void accumulate_weighted_avg(vw& all, regressor& reg)
+void accumulate_weighted_avg(vw& all, weight_parameters& weights)
 { if(!all.adaptive)
   { cerr<<"Weighted averaging is implemented only for adaptive gradient, use accumulate_avg instead\n";
     return;
   }
   uint32_t length = 1 << all.num_bits; //This is the number of parameters
-  size_t stride = 1 << all.reg.stride_shift;
-  weight* weights = reg.weight_vector;
-  float* local_weights = new float[length];
+  weight_parameters local_weights(length);
 
-  for(uint32_t i = 0; i < length; i++)
-    local_weights[i] = weights[stride*i+1];
+  weight_parameters::iterator i = weights.begin(1);
+  weight_parameters::iterator j = local_weights.begin();
```

Comment:
It's not clear to me that local_weights should be a weight_parameters because it has a rather different meaning.  This is temporary storage just for the purpose of communication.